src/cdk/commands/aslVisualizationCDKManager.ts

```diff
@@ -0,0 +1,94 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+
+import { AslVisualizationCDK } from './aslVisualizationCDK'
+import { ConstructNode } from '../explorer/nodes/constructNode'
+import { getLogger, Logger } from '../../shared/logger'
+import { AbstractAslVisualizationManager } from '../../../src/stepFunctions/commands/visualizeStateMachine/abstractAslVisualizationManager'
+import { StateMachineGraphCache } from '../../../src/stepFunctions/utils'
+
+export class AslVisualizationCDKManager extends AbstractAslVisualizationManager {
+    protected readonly managedVisualizations: Map<string, Map<string, AslVisualizationCDK>> = new Map<string, Map<string, AslVisualizationCDK>>()
+
+    public constructor(extensionContext: vscode.ExtensionContext) {
+        super(extensionContext)
+    }
+
+    public async visualizeStateMachine(
+        globalStorage: vscode.Memento,
+        node: ConstructNode
+    ): Promise<vscode.WebviewPanel | undefined> {
+        const logger: Logger = getLogger()
+        const cache = new StateMachineGraphCache()
+        const stateMachineName = node.label
+        const cdkOutPath = node.id?.replace(`/tree.json/${node.tooltip}`, ``)
+        const appName = node.tooltip?.replace(`/${stateMachineName}`, ``)
+        const templatePath = String(cdkOutPath) + `/${appName}.template.json`
+        const uri = vscode.Uri.file(templatePath);
+        const workspaceName = this.getCDKAppWorkspaceName(cdkOutPath!)
+        const existingVisualization = this.getExistingVisualization(workspaceName, node.tooltip!)
+        if (existingVisualization) {
+            existingVisualization.showPanel()
+
+            return existingVisualization.getPanel()
+        }
+
+        // Existing visualization does not exist, construct new visualization
+        try {
+            await cache.updateCache(globalStorage)
+
+            const textDocument = await vscode.workspace.openTextDocument(uri)
+            const newVisualization = new AslVisualizationCDK(textDocument, templatePath, appName!, stateMachineName)
+            if (newVisualization) {
+                this.handleNewVisualization(workspaceName, newVisualization)
+                return newVisualization.getPanel()
+            }
+        } catch (err) {
+            this.handleErr(err, logger)
+        }
+
+        return
+    }
+
+    protected handleNewVisualization(workspaceName: string, newVisualization: AslVisualizationCDK): void {
+        let map = this.managedVisualizations.get(workspaceName)
+        const uniqueIdentifier = newVisualization.cdkAppName + '/' + newVisualization.stateMachineName
```

Comment:
Is this guaranteed unique?

Reply:
Yes it is guaranteed unique, because it is impossible to have two cdk applications with the same name within a workspace or have two state machines with the same name within a cdk application.